nemo/collections/asr/modules/rnnt.py

```diff
@@ -1078,6 +1078,32 @@ def mask_select_states(
         # LSTM in PyTorch returns a tuple of 2 tensors as a state
         return states[0][:, mask], states[1][:, mask]
 
+    def batch_copy_states_mask(
```

Comment:
It's a bit strange, but basically mask_select_states does a host<->device synchronization. `states[0][:, mask]` creates a smaller tensor whose shape is determined by the number of True values in mask. This ends up doing a device to host copy of the number of 1's, so that a memory allocation can be done on host.

batch_copy_states does something similar, since indexing with the integer tensor `ids` also creates a new allocation. 

This is different from what I end up doing, which just rewrites some values based on the mask using torch.where()

Anyway I need to make an abstract method and implement this for the convolution-based prediction networks as well.